Week1

dpezelj opened the pull request:
Week 1 Assignments.

dpezelj committed:
Added MUI
Added Roboto font to index.html as basic font
Added Theme to App.tsx
Added Navigation component

dpezelj committed:
Added Header component with bg image and titles

dpezelj committed:
Added preparation for Footer

dpezelj committed:
Added Footer.tsx

dpezelj committed:
Added CityCard component

dpezelj committed:
Added Accommodation Card component
Fixed Footer width being out of screen

dpezelj committed:
Added PlaceCard component

dpezelj committed:
Added AccommodationInfo component

dpezelj committed:
Refactor imports

dpezelj committed:
Update AccommodationInfo component - props

dpezelj committed:
Add data type

antonioorct commented on src/components/AccommodationCard.tsx:
```diff
@@ -0,0 +1,93 @@
+            EUR {price}
+          </Typography>
+          <Stack direction="row" spacing={1}>
+            <div style={stars}></div>
```
Instead of using a div with a style of the star, maybe you could try making the Star svg as a react component so it's Star.tsx returning the svg and then calling it here

antonioorct replied:
Even better solution that came to my mind now is to create a `Rating` component that takes a number of stars you wish to show and takes care of all of display the desired amount of stars. You don't have to do this now but would be a better solution here in the future but also in general

antonioorct commented on src/components/AccommodationInfo.tsx:
```diff
@@ -0,0 +1,196 @@
+              maxWidth: "810px",
+            }}
+          >
+            This property is 3 minutes walk from the beach. Overlooking Mykonos
+            Windmills, the Poseidon Hotel Suites is only 50 m from Megali Ammos
+            Beach. The 3-star hotel offers a freshwater pool, and bright rooms
+            with air conditioning and fan. Each of the Cycladic rooms opens to a
+            private balcony with across to Mykonos Town, the sea and Delos. A
+            fridge, satellite TV and safe are standard. Free two-way transfer
+            from the airport or port is offered. Poseidon provides free Wi-Fi in
+            public areas, and on-site parking is also free. Guests can hire
+            Poseidon’s motor yacht and discover the magnificent beaches of
+            Mykonos. At the breakfast room and its cool patio, guests can taste
+            homemade local delicacies, fresh fruit and good quality coffee. The
+            Alley Bay serves exclusive cocktails, a few steps from the Poseidon.
+            The Poseidon is just 200 m from the famous Mykonos Windmills. Right
+            next to the hotel, guests will find small sandy coves and a pathway
+            that leads to the picturesque chapel of Agios Charalambis. This is
+            our guests' favourite part of Mýkonos City, according to independent
+            reviews. Couples particularly like the location — they rated it 9.6
+            for a two-person trip.
```
Pass down the description as a prop

antonioorct commented on src/components/CityCard.tsx:
```diff
@@ -0,0 +1,61 @@
+    lineHeight: "24px",
+  };
+
+  let formatCount = Number(count).toLocaleString();
```
No need to cast `count` to a number, just declare the `count` property as a number instead of string in `CityCardProps`

antonioorct commented on src/data/data.js:
```diff
@@ -0,0 +1,40 @@
+
+const cityCardData = {
+    name: "London",
+    count: "5102",
```
I can see why you've declared the `count` property in `CityCardProps` as a string. It just makes more sense for the `count` property to be a number. So just use a number here instead of this string

antonioorct commented on src/data/data.js:
```diff
@@ -0,0 +1,40 @@
+
```
You can keep this as a .js file but in the future if you want you can convert this to .ts and declare the corresponding types

antonioorct commented on src/components/Navigation.tsx:
```diff
@@ -0,0 +1,70 @@
+    padding: "0px 76px",
+  };
+
+  const menuItems = [" Locations", "My Places", "My Bookings"];
```
This is a hard coded value and if it is always the same value (it is), you can extract it outside of this component or into another file (constants.ts)

dpezelj committed:
Add Rating.tsx component
Update some pieces of code to look more like MUI
 Fixed problems with key attr inside components in Navigation and Rating
